src/main/java/ch/njol/skript/expressions/ExprValueWithin.java

```diff
@@ -0,0 +1,72 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.WrapperExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Value Within")
+@Description("Gets the value within objects. Usually used with variables to get the value they store rather than the variable itself, " +
+	"or with lists to get the values of a type.")
+@Examples({
+	"set {_entity} to a random entity out of all entities",
+	"delete entity within {_entity} # This deletes the entity itself and not the value stored in the variable",
+	"",
+	"set {_list::*} to \"something\", 10, \"test\" and a zombie",
+	"broadcast the strings within {_list::*} # \"something\", \"test\""
+})
+@Since("INSERT VERSION")
+public class ExprValueWithin extends WrapperExpression<Object> {
+
+	static {
+		Skript.registerExpression(ExprValueWithin.class, Object.class, ExpressionType.SIMPLE, "[the] (%-*classinfo%|value[s]) (within|in) %objects%");
```

Comment:
you probably shouldn't be able to do `the values within {_single value}` or `the numbers within {_single value}` which I think this would allow. You also shouldn't be able to do `the number within {_list of multiple numbers::*}`, as there may be more than one number.

Reply:
You can't tell if the given classinfo was a plural or not, and I don't think we want to use regex and parse the classinfo our own just to know if the input was plural or not

Reply:
You can check in the init method if the provided classinfo was a plural or not. You did so in your other pull request https://github.com/SkriptLang/Skript/blob/37cbae397d6131098db3dd976652cdb6b8dd7089/src/main/java/ch/njol/skript/expressions/ExprItems.java#L79-L81

Reply:
> You did so in your other pull request
That's different because the pattern ended with the classinfo expression, this on the other hand has the classinfo expression in the middle, so it's harder to get the raw string used for the classinfo

Reply:
Could substring after second space character

Reply:
Where would the substring end, like substring to up to where? We can't know where the type ends as it could contain multiple words e.g. `living entity` and an addon could have a type that contains `in`/`within` and we would have no idea if it was a part of the type or the pattern

Also, why substring after the second space

Reply:
`in/within` wouldn't come up as a plural

Reply:
You can also use tags on the `[:the]` to determine if you need to substring again.